plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/transformers/ProxyReplacerTransformer.java

```diff
@@ -39,27 +41,46 @@ private static CtMethod overrideMethod(CtClass ctClass, CtMethod getConnectionMe
     }
 
     /**
-     *
      * @param ctClass
      * @throws NotFoundException
      * @throws CannotCompileException
      */
-    @OnClassLoadEvent(classNameRegexp = "org.springframework.beans.factory.support.DefaultListableBeanFactory")
+    @OnClassLoadEvent(classNameRegexp = "org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory")
     public static void replaceBeanWithProxy(CtClass ctClass) throws NotFoundException, CannotCompileException {
         CtMethod[] methods = ctClass.getMethods();
         for (CtMethod ctMethod : methods) {
```

Comment:
It's not necessary to loop all the method , maybe we just use ctClass.getDeclaredMethod(FACTORY_CREATE_METHOD_NAME, new CtClass[] {xxx}).

Reply:
done